pcmanfm/mainwindow.cpp

```diff
@@ -260,6 +267,28 @@ void MainWindow::addTab(FmPath* path) {
   }
 }
 
+void MainWindow::toggleMenuBar() {
+  Settings& settings = static_cast<Application*>(qApp)->settings();
+  bool showMenuBar = !settings.showMenuBar();
+
+  if (!showMenuBar) {
+    QScopedPointer<QMessageBox> warning(
```

Comment:
Why are you creating the `QMessageBox` widget on the heap and not on the the stack?